streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,37 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    // Test for success with zero-argument constructor and sparkConf constructor
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners",
+        classOf[StreamingListenerThatAcceptsSparkConf].getName + "," +
+        classOf[ReceiverInfoCollector].getName)
+    val scc = new StreamingContext(conf, Seconds(1))
```

Comment:
This can lead to a leak in the StreamingContext and underlying SparkContext. Please put the whole in 

```
var ssc: StremaingContext = null
try {
  ssc = new StreamingContext...

} finally {
   if (ssc != null) {
     ssc.stop()
   }
}

```